CONTRIBUTING.rst

```diff
@@ -0,0 +1,500 @@
+   pull request for the appropriate branch. The title and message from your
+   commit message should appear as well.
+
+#. If you're working on a subsystem branch that's not ``master``,
+   you may need to change the intended branch for the pull-request
+   here, for example, by changing the base branch from ``master`` to ``net``.
+
+#. GitHub will assign one or more suggested reviewers (based on the
+   CODEOWNERS file in the repo) and you can also select additional reviewers
+   now too.
+
+#. Click on the submit button and your pull-request is sent and awaits
+   review.  Email will be sent as review comments are made, or you can check
+   on your pull-request at https://github.com/zephyrproject-rtos/zephyr/pulls.
+
+#. While you're waiting for your pull-request to be accepted and merged, you
+   can create another branch to work on another issue. (Be sure to make your
+   new branch off of master and not the previous branch.)::
+
+     $ git checkout master
+     $ git checkout -b fix_another_issue
+
+   and use the same process described above to work on this new topic branch.
+
+#. If reviewers do request changes to your patch, go back to your local repo,
+   verify you're on the right branch, make changes, ammend your commit, rebase
+   to sync with the upstream master branch (avoiding merge conflicts), and
+   force push your changes::
+
+     $ git status
+     $ git checkout fix_comment_typo
+
+      <make recommend edits from reviewers>
+
+     $ git commit --amend     # instead of creating another commit
+
+     $ git fetch upstream
+     $ git rebase upstream/master
+
+     $ git push -f origin fix_comment_typo
+
+   By force pushing your update, your original pull-request will be updated
+   with your new change.
+
+
+GitHub Tips and Tricks
+**********************
+
+Rebasing commits to fix code review issues
+==========================================
+
+You can interactively rebase offending commit(s) to fix code review
+issues.  In your development repo::
+
+  $ git fetch --all
+  $ git rebase --ignore-whitespace origin master
+  $ git rebase -i <offending-commit-id>^
+
+.. note::
+
+   The ``--ignore-whitespace``option stops ``git apply`` (which is
+   called by rebase) from changing any whitespace.
+
+Replace ``pick`` with ``edit`` or remove the line to delete a commit.
+Fix the issue in the code review::
+
+  $ git add [file(s)]
+  $ git rebase --continue
+
+  <update commit comment if needed>
+
+  $ git push --force origin fix_comment_typo
+
+.. _Commit Guidelines:
+
+Commit Guidelines
+*****************
+
+Changes are submitted as Git commits. Each commit message must contain:
+
+* A short and descriptive subject line that is less than 72 characters,
+  followed by a blank line. The subject line must include a prefix that
+  identifies the subsystem being changed, folowed by a colon, and a short
+  title, for example:  ``doc: update wiki references to new site``
```

Comment:
Might be worth recommending "git log $FILE_YOU_TOUCHED" to get examples of what other people have used as a prefix, for those in doubt.